### ISSUE TITLE
blocking number prefixes

### ISSUE THREAD

**Issue by CaseOf** [x13a/Silence#254]
Hello,

Actually, neither dialer or Silence can block callers from defined number prefixes.
It would be nice if you can add this feature.

If you want an exemple usage, French government did force sales prospection callers by law to use defined number prefixes that are public : https://www.quechoisir.org/actualite-demarchage-telephonique-interdit-a-partir-de-numeros-de-telephone-mobile-n105035/

It would help to easily block all prospection calls in this example country (where I live).

**Comment by emheyarssi** [comment]
Try to use the regex settings.

**Comment by CaseOf** [comment]
Do you mean in stock dialer app? 
Silence does not give the ability to add custom numbers.

**Comment by emheyarssi** [comment]
Yes, sorry, I use a modified version from the repository. In my case, I allow number starting with a given prefix (random number from voip providers in pacific). And refuse all but contacts list from phone.
